src/tail/log_server.rs

```diff
@@ -0,0 +1,63 @@
+use hyper::server::conn::AddrIncoming;
+use hyper::server::Builder;
+use hyper::service::{make_service_fn, service_fn};
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+use tokio::sync::oneshot::Receiver;
+
+pub struct LogServer {
+    server: Builder<AddrIncoming>,
+    shutdown_rx: Receiver<()>,
+}
+
+/// LogServer is just a basic HTTP server running locally; it listens for POST requests on the root
+/// path and simply prints the JSON body of each request as its own line to STDOUT.
+impl LogServer {
+    pub fn new(shutdown_rx: Receiver<()>) -> LogServer {
+        // Start HTTP echo server that prints whatever is posted to it.
+        let addr = ([127, 0, 0, 1], 8080).into();
```

Comment:
8080 is common port and we expect tail to run for extended periods of time. Should we make this a less common port or give users to option to change port?

Reply:
there is an issue to allow port configuration #1176